scripts/bump_deps.lua

```diff
@@ -1,343 +1,449 @@
-		pr_body = relevant_commit:gsub(commit_prefix:gsub("%(", "%%("):gsub("%)", "%%)"), "")
-	end
-	pr_body = pr_body .. "\n\n(add explanations if needed)"
-	p(pr_title .. "\n" .. pr_body .. "\n")
-	create_pr(pr_title, pr_body)
+  require_executable('git')
+
+  verify_branch('deps')
+
+  local nvim_remote = find_git_remote(nil)
+  local relevant_commit = assert(run_die({
+    'git',
+    'log',
+    '--grep=' .. commit_prefix,
+    '--reverse',
+    "--format='%s'",
+    nvim_remote .. '/master..HEAD',
+    '-1',
+  }, 'Failed to fetch commits'))
+
+  local pr_title
+  local pr_body
+
+  if relevant_commit == '' then
+    pr_title = commit_prefix .. 'bump some dependencies'
+    pr_body = 'bump some dependencies'
+  else
+    relevant_commit = relevant_commit:gsub("'", '')
+    pr_title = relevant_commit
+    pr_body = relevant_commit:gsub(commit_prefix:gsub('%(', '%%('):gsub('%)', '%%)'), '')
+  end
+  pr_body = pr_body .. '\n\n(add explanations if needed)'
+  p(pr_title .. '\n' .. pr_body .. '\n')
+  create_pr(pr_title, pr_body)
+end
+
+local function usage()
+  local this_script = debug.getinfo(1,'S').source:match("[^/]*.lua$")
+  print(([=[
+    Bump Nvim dependencies
+
+    Usage:  nvim -l %s [options]
+        Bump dep to a tagged version:
+            nvim -l scripts/bump_deps.lua --dep Luv --version 1.43.0-0
+        Bump to HEAD:
+            nvim -l scripts/bump_deps.lua --dep Luv --head
+        Bump to commit:
+            nvim -l scripts/bump_deps.lua --dep Luv --commit abc123
+        Create a branch:
+            nvim -l scripts/bump_deps.lua --dep Luv --branch
+        Create a PR:
+            nvim -l scripts/bump_deps.lua --pr
+
+    Options:
+        -h                    show this message and exit.
+        --pr                  submit pr for bumping deps.
+        --branch <dep>        create a branch bump-<dep> from current branch.
+        --dep <dependency>    bump to a specific release or tag.
+
+    Dependency Options:
+        --version <tag>       bump to a specific release or tag.
+        --commit <hash>       bump to a specific commit.
+        --HEAD                bump to a current head.
+
+        <dependency> is one of:
+        \"LuaJIT\", \"libuv\", \"Luv\", \"tree-sitter\"
+  ]=]):format(this_script))
 end
 
-return M
+local function parseargs()
+  local args = {}
+  for i = 1, #_G.arg do
+    if _G.arg[i] == '-h' then
+      args.h = true
+    elseif _G.arg[i] == '--pr' then
+      args.pr = true
+    elseif _G.arg[i] == '--branch' then
+      args.branch = _G.arg[i+1]
+    elseif _G.arg[i] == '--dep' then
+      args.dep = _G.arg[i+1]
+    elseif _G.arg[i] == '--version' then
+      args.version = _G.arg[i+1]
+    elseif _G.arg[i] == '--commit' then
+      args.commit = _G.arg[i+1]
+    elseif _G.arg[i] == '--head' then
+      args.head = true
+    end
+  end
+  return args
+end
+
+local scriptname = vim.v.argv[#vim.v.argv - #_G.arg]
+local is_main = scriptname:match('bump_deps.lua')
```

Comment:
Need some sort of `vim.is_main()` (?) thing that scripts can use to decide whether to run like a script or like a module. Similar to python `__name__ == "__main__"` pattern.